Samples/WinMLSamplesGallery/TransformAsync/SegmentModel.cpp

```diff
@@ -0,0 +1,279 @@
+#include "pch.h"
+#include "SegmentModel.h"
+#include <iostream>
+#include <filesystem>
+
+using winrt::Windows::Foundation::PropertyValue;
+using winrt::hstring;
+using namespace winrt;
+using namespace Windows::Foundation::Collections;
+
+enum OnnxDataType : long {
+	ONNX_UNDEFINED = 0,
+	// Basic types.
+	ONNX_FLOAT = 1,
+	ONNX_UINT8 = 2,
+	ONNX_INT8 = 3,
+	ONNX_UINT16 = 4,
+	ONNX_INT16 = 5,
+	ONNX_INT32 = 6,
+	ONNX_INT64 = 7,
+	ONNX_STRING = 8,
+	ONNX_BOOL = 9,
+
+	// IEEE754 half-precision floating-point format (16 bits wide).
+	// This format has 1 sign bit, 5 exponent bits, and 10 mantissa bits.
+	ONNX_FLOAT16 = 10,
+
+	ONNX_DOUBLE = 11,
+	ONNX_UINT32 = 12,
+	ONNX_UINT64 = 13,
+	ONNX_COMPLEX64 = 14,     // complex with float32 real and imaginary components
+	ONNX_COMPLEX128 = 15,    // complex with float64 real and imaginary components
+
+	// Non-IEEE floating-point format based on IEEE754 single-precision
+	// floating-point number truncated to 16 bits.
+	// This format has 1 sign bit, 8 exponent bits, and 7 mantissa bits.
+	ONNX_BFLOAT16 = 16,
+}OnnxDataType;
+
+
+const int32_t opset = 12;
+
+/****	Style transfer model	****/
+void StyleTransfer::InitializeSession(int w, int h)
+{
+	SetImageSize(720, 720); // SIze model input sizes fixed to 720x720
+	m_session = CreateLearningModelSession(GetModel());
+	m_binding = LearningModelBinding(m_session);
+}
+void StyleTransfer::Run(IDirect3DSurface src, IDirect3DSurface dest)
+{
+	m_syncStarted = TRUE;
+
+	VideoFrame inVideoFrame = VideoFrame::CreateWithDirect3D11Surface(src);
+	VideoFrame outVideoFrame = VideoFrame::CreateWithDirect3D11Surface(dest);
+	SetVideoFrames(inVideoFrame, outVideoFrame);
+
+	hstring inputName = m_session.Model().InputFeatures().GetAt(0).Name();
+	m_binding.Bind(inputName, m_inputVideoFrame);
+	hstring outputName = m_session.Model().OutputFeatures().GetAt(0).Name();
+
+	auto outputBindProperties = PropertySet();
+	outputBindProperties.Insert(L"DisableTensorCpuSync", PropertyValue::CreateBoolean(true));
+
+	m_binding.Bind(outputName, m_outputVideoFrame, outputBindProperties); 
+	auto results = m_session.Evaluate(m_binding, L"");
+
+	m_outputVideoFrame.CopyToAsync(outVideoFrame).get();
+
+	m_syncStarted = FALSE;
+}
+
+LearningModel StyleTransfer::GetModel()
+{
+	auto model_path = std::filesystem::current_path();
+	model_path.append("Assets\\mosaic.onnx");
+	return LearningModel::LoadFromFilePath(model_path.c_str());
+}
+
+/****	Background blur model	****/
+BackgroundBlur::~BackgroundBlur() 
+{
+	if (m_session) m_session.Close();
```

Comment:
>Close() [](http://example.com/codeflow?start=26&length=7)

Same question here - do we need to explicitly close it, given the destructor should?

Reply:
not needed because inherits the base destructor